react.js

```diff
@@ -101,16 +136,195 @@ export function useRef(value) {
   }
 }
 
-export function useContext() {
+export function useContext(context) {
+  if (!currentStore) {
+    throw new Error('Hook called not in a component');
+  }
+
+  let s = currentStore;
+  let store = get(currentStore);
+  let index = i++;
+
+  if (store[index] === undefined) {
+    let initial = true;
+    context.__store.subscribe(value => {
+      store[index] = value;
+      if (initial) {
+        initial = false;
+      } else {
+        s.set(store);
+      }
+    });
+  }
 
+  return store[index];
 }
 
 export function useCallback(cb, deps) {
   return useMemo(() => cb, deps);
 }
 
-export function createContext() {
-  return {};
+export function createContext(initialValue) {
+  let store = writable(initialValue);
+  const key = Symbol();
+  // Context.Provider
+  /*
+    <script>
+      import { setContext } from 'svelte';
+      import { writable } from 'svelte/store';
+      
+      export let value = initialValue;
+      let store = writable(value);
+
+      setContext(key, store);
+      
+      $: store.set(value)
+    </script>
+
+    <slot />
+  */
+  function create_provider_fragment(ctx) {
```

Comment:
Hmm is this safe to do? Seems like we're relying on a bunch of compiler internals here. What happens if they change?

Maybe we could have a general purpose Provider and Consumer component that have a `this.key` property. Then `createContext` could extend those component classes and assign `this.key` to the proper key. Might avoid copy pasting compiled code into here potentially?

Reply:
hmm... interesting idea. but i dont think its possible to get `this` from within a svelte component. 🤔

Reply:
Ah yeah...

Reply:
Maybe key could be a prop, and the extended class could call super and add the key to the options?

Reply:
oh i was thinking about the same thing! updated the PR :)